src/link_manager/LinkManagerStateMachine.cpp

```diff
@@ -706,28 +712,41 @@ void LinkManagerStateMachine::handleSwssLinkStateNotification(const link_state::
 //
 void LinkManagerStateMachine::handleMuxConfigNotification(const common::MuxPortConfig::Mode mode)
 {
-    if (mComponentInitState.all()) {
-        if (mode == common::MuxPortConfig::Mode::Active &&
-            ms(mCompositeState) != mux_state::MuxState::Label::Active &&
-            ms(mCompositeState) != mux_state::MuxState::Label::Wait) {
-            CompositeState nextState = mCompositeState;
-            enterLinkProberState(nextState, link_prober::LinkProberState::Wait);
-            switchMuxState(nextState, mux_state::MuxState::Label::Active);
-            LOGWARNING_MUX_STATE_TRANSITION(mMuxPortConfig.getPortName(), mCompositeState, nextState);
-            mCompositeState = nextState;
-        } else if(mode == common::MuxPortConfig::Mode::Standby &&
-                  ms(mCompositeState) != mux_state::MuxState::Label::Standby &&
-                  ms(mCompositeState) != mux_state::MuxState::Label::Wait) {
-            mSendPeerSwitchCommandFnPtr();
-        } else {
-            mMuxStateMachine.setWaitStateCause(mux_state::WaitState::WaitStateCause::DriverUpdate);
-            mMuxPortPtr->probeMuxState();
+    if (mComponentInitState.test(MuxStateComponent) &&
+        mode != common::MuxPortConfig::Mode::Auto && 
+        mode != common::MuxPortConfig::Mode::Manual &&
+        ms(mCompositeState) == mux_state::MuxState::Wait) {
+        
+        MUXLOGWARNING(boost::format("%s: Mux mode: %s , mux mode config change is pending. ") % 
+                mMuxPortConfig.getPortName() %
+                mMuxStateName[ms(mCompositeState)]
+            );
+
+        mPendingMuxModeChange = true;
+        mTargetMuxMode = mode;
+    } else {
```

Comment:
Suggest always record the pending action first. And make the log level notice. (code consistency)

Then call the helper function if the state is right below.

Add a matching notice log in the helper function so that we know when the request was executed.